SonarQube.TeamBuild.PreProcessor/ArgumentProcessor.cs

```diff
@@ -19,6 +20,12 @@ namespace SonarQube.TeamBuild.PreProcessor
     /// </summary>
     public static class ArgumentProcessor // was internal
     {
+        /// <summary>
+        /// Regular expression to validate a project key.
+        /// See http://docs.sonarqube.org/display/SONAR/Project+Administration#ProjectAdministration-AddingaProject
+        /// </summary>
+        private static readonly Regex ProjectKeyRegEx = new Regex(@"^[0-9a-zA-Z:-_\.]+$", RegexOptions.Compiled | RegexOptions.Singleline);
+
```

Comment:
@dbolkensteyn  I copied the regular expression from the SonarQube doc link list in line 25.
However, I'm not sure how the expression ":-_" is supposed to be interpreted. Strictly speaking it means "all of the characters between : and _", which includes A to Z, which are have already been explicitly included in the expression.
Do you know if that was what was intended, or did it mean that : - and _ should be treated as valid characters?